istioctl/cmd/auth.go

```diff
@@ -234,16 +237,73 @@ func getConfigDumpFromPod(podName, podNamespace string) (*configdump.Wrapper, er
 }
 
 func newConverter(v1PolicyFiles, serviceFiles []string, istioNamespace, istioMeshConfigMapName string) (*converter.Converter, error) {
-	if len(v1PolicyFiles) == 0 {
-		return nil, fmt.Errorf("no input file provided")
+	v1alpha1Policies, err := getV1alpha1Policies(v1PolicyFiles)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get v1alpha1 policies: %v", err)
 	}
 
-	var k8sClient *kubernetes2.Clientset
 	k8sClient, err := kube.CreateClientset("", "")
 	if err != nil {
 		return nil, fmt.Errorf("failed to connect to Kubernetes: %v", err)
 	}
-	return converter.New(k8sClient, v1PolicyFiles, serviceFiles, meshConfig, istioNamespace, istioMeshConfigMapName)
+	return converter.New(k8sClient, v1alpha1Policies, serviceFiles, meshConfig, istioNamespace, istioMeshConfigMapName)
+}
+
+func getV1alpha1Policies(v1PolicyFiles []string) (*model.AuthorizationPolicies, error) {
```

Comment:
I think auth.go is the interface, so it's better to actually getting the policies in converter, unless this function is shared then we can make this as a util function.

Reply:
It's better to put this in `istioctl/cmd/auth.go`. This function is not shared with other code and that's one of the reason I think it shouldn't go to `converter.go`.

The other reason is the `getV1alpha1Policies()` uses some code that is only available in istioctl, for example, the `clientExecFactory()`.

This also means we probably should also refactor to not to pass `k8sClient` and create a better data structure to pass the service information instead of passing the files directly.

Reply:
I still think it's better to move `getV1alpha1Policies` to somewhere else since it's only for the conversion tool and since this file is more of a high level interface of the cmd, so let's keep it simple. We can make `clientExecFactory()` a public function since it's quite useful to use in some other places outside of the `cmd` package or you can pass it to the `getV1alpha1Policies` function. This can be a TODO if you think it makes sense.

Reply:
> I still think it's better to move getV1alpha1Policies to somewhere else since it's only for the conversion tool and since this file is more of a high level interface of the cmd, so let's keep it simple.

Isn't what you said the exact reason to keep the getV1alpha1Policies right in the `auth.go`? 

> We can make clientExecFactory() a public function since it's quite useful to use in some other places outside of the cmd package or you can pass it to the getV1alpha1Policies function. This can be a TODO if you think it makes sense.

I think we should restrict the using of clientExecFactory to be inside istioctl and avoid using it in anywhere else.